solr/core/src/java/org/apache/solr/handler/component/QueryComponent.java

```diff
@@ -288,6 +328,13 @@ protected void prepareGrouping(ResponseBuilder rb) throws IOException {
     groupingSpec.setMain(params.getBool(GroupParams.GROUP_MAIN, false));
     groupingSpec.setNeedScore((rb.getFieldFlags() & SolrIndexSearcher.GET_SCORES) != 0);
     groupingSpec.setTruncateGroups(params.getBool(GroupParams.GROUP_TRUNCATE, false));
+
+    groupingSpec.setSkipSecondGroupingStep(params.getBool(GroupParams.GROUP_SKIP_DISTRIBUTED_SECOND, false));
+    boolean isReranking = (rb.getRankQuery() != null);
+    if (groupingSpec.isSkipSecondGroupingStep() & !allowSkipSecondGroupingStep(groupingSpec.getWithinGroupSortSpec(), groupingSpec.getGroupSortSpec(), isReranking)){
```

Comment:
observation: `allowSkipSecondGroupingStep` currently does 'error' logging in response to illegal grouping specifications in a request. this probably should not be 'error' or 'warn' logging (since there is nothing 'wrong' with Solr itself) but 'info' logging at most or no logging at all with the details returned in the `Illegal grouping specification` that is thrown here. An alternative could also be to 'info' log and _not_ throw but to just ignore the `group.skip.second.step=true` parameter if it's invalid.

Reply:
done in fc8c9c7

Reply:
The `QueryComponent.allowSkipSecondGroupingStep` method being passed and (almost) only using two of the `GroupingSpecification`'s parameters made me wonder if the allow logic might be moved into a `validate()` or similarly named method in that class -- https://github.com/cpoerschke/lucene-solr/commit/70e9c19ca5584be5c0666f70efea25b368a6f3b8 has a sketch -- the logic being there would make it potentially available also to other non-`QueryComponent` plugins and the explicit 'optionally call validate after all the setters' aspect might also be helpful in terms of code comprehension. What do you think?

Reply:
I like it, but what about `validate` being `void` and throwing  a `SolrException` in case of problems? in this case we can return the description of the particular problem in the exception and we just call `validate` at the end of the setup. what do you think ?

Reply:
Sure, `void` and throwing would work too, good idea, as you say more details can be returned that way.

Reply:
Done in 0563e6618a3c5f049388583cd558fc08af684db8

Reply:
> edit: a little further below in the class there's a cmd.setSegmentTerminateEarly(false); // not supported, silently ignore any segmentTerminateEarly flag for example

Checking for this and failing if it is true will modify the current  behavior  - do you think we should fail?